src/NetMQ.Tests/CurveTests.cs

```diff
@@ -0,0 +1,100 @@
+using System.Text;
+using NetMQ.Sockets;
+using Xunit;
+
+#if NET47
+using ZeroMQ;
+#endif
+
+namespace NetMQ.Tests
+{
+    public class CurveTests
+    {
+        [Fact]
+        public void CurveTest()
+        {
+            var serverPair = new NetMQCertificate();
+            using var server = new DealerSocket();
+            server.Options.CurveServer = true;
+            server.Options.CurveCertificate = serverPair;
+            server.Bind($"tcp://127.0.0.1:55367");
+            
+            var clientPair = new NetMQCertificate();
+            using var client = new DealerSocket();
+            client.Options.CurveServerKey = serverPair.PublicKey;
+            client.Options.CurveCertificate = clientPair;
+            client.Connect("tcp://127.0.0.1:55367");
```

Comment:
@somdoron  I would like to use curve encryption in a scenario where i need to ensure integrity, authenticity and confidentiality. For what I understood in this example/test only server identity is verified, is it possible to verify also clients'? In fact, in my scenario there isn't a client-server architecture but clients/peers communicating with each other.

Thanks in advance

Reply:
yes, however that is not implemented yet.
We need to port ZAP from libzmq for that. I will take some more time.

Reply:
You can develop your own protocol on top. Either with username and password or EC signing.

Reply:
Thanks